source/extensions/stat_sinks/open_telemetry/open_telemetry_impl.cc

```diff
@@ -0,0 +1,171 @@
+#include "source/extensions/stat_sinks/open_telemetry/open_telemetry_impl.h"
+
+#include "source/common/tracing/null_span_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace OpenTelemetry {
+
+OtlpOptions::OtlpOptions(const SinkConfig& sink_config)
+    : report_counters_as_deltas_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, report_counters_as_deltas, false)),
+      report_histograms_as_deltas_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, report_histograms_as_deltas, false)),
+      emit_tags_as_attributes_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, emit_tags_as_attributes, true)),
+      use_tag_extracted_name_(
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(sink_config, use_tag_extracted_name, true)) {}
+
+OpenTelemetryGrpcMetricsExporterImpl::OpenTelemetryGrpcMetricsExporterImpl(
+    const OtlpOptionsSharedPtr config, Grpc::RawAsyncClientSharedPtr raw_async_client)
+    : config_(config), client_(raw_async_client),
+      service_method_(*Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
+          "opentelemetry.proto.collector.metrics.v1.MetricsService.Export")) {}
+
+void OpenTelemetryGrpcMetricsExporterImpl::send(MetricsExportRequestPtr&& export_request) {
+  if (export_request == nullptr) {
+    return;
+  }
+
+  client_->send(service_method_, *export_request, *this, Tracing::NullSpan::instance(),
+                Http::AsyncClient::RequestOptions());
+}
+
+void OpenTelemetryGrpcMetricsExporterImpl::onSuccess(
+    Grpc::ResponsePtr<MetricsExportResponse>&& export_response, Tracing::Span&) {
+  if (export_response->has_partial_success()) {
+    ENVOY_LOG(warn, "export response with partial success; {} rejected, collector message: {}",
+              export_response->partial_success().rejected_data_points(),
+              export_response->partial_success().error_message());
```

Comment:
This could cause significant log spam. Either drop this to debug or use a backoff log. Optimally this kind of thind would be covered by a stat.

Reply:
done